dashel/dashel-win32.cpp

```diff
@@ -933,13 +944,66 @@ namespace Dashel
 			}
 
 			hf = CreateFileA(devName.c_str(), GENERIC_WRITE | GENERIC_READ, 0, NULL, OPEN_EXISTING, FILE_FLAG_OVERLAPPED, NULL);
-			if(hf == INVALID_HANDLE_VALUE)
-				throw DashelException(DashelException::ConnectionFailed, GetLastError(), "Cannot open serial port.");
+			if (hf == INVALID_HANDLE_VALUE)
+				throw DashelException(DashelException::ConnectionFailed, GetLastError(),
+					(std::string("Cannot open serial port ") + devName + ".").c_str());
 
 			buildDCB(hf, target.get<int>("baud"), target.get<int>("bits"), target.get<bool>("dtr"), target.get("parity"), target.get("stop"), target.get("fc"));
 
 			startStream(EvPotentialData);
 		}
+
+		virtual bool checkConnection() const
```

Comment:
I see. In general, the design decision behind my software is to have the least possible virtuals, at least in non-public API. As this is a private API (SerialStream is never exposed to the user of the library), I suggest to keep it non-virtual, as we could just add the virtual in the hypothetical case we want to override it.